torch/_dynamo/eval_frame.py

```diff
@@ -262,19 +262,61 @@ def innermost_fn(fn):
     return unaltered_fn
 
 
+# The config to restore to should dynamo compile / recompile when
+# executing from the compiled function's _TorchDynamoContext
+config_cache = threading.local()
+
+
+def _maybe_init_guarded_config_cache():
+    if not hasattr(config_cache, "saved_config_and_hash"):
+        config_cache.saved_config_and_hash: Optional[  # type: ignore[misc]
+            Tuple[Dict[str, Any], bytes]
+        ] = None
+
+
 @contextlib.contextmanager
-def enable_dynamic(enable: Optional[bool] = None, export: bool = False):
-    if enable is None:
+def restore_guarded_dynamo_config(
+    first_ctx: bool,
+    saved_config: Dict[str, Any],
+    saved_config_hash: str,
+):
+    _maybe_init_guarded_config_cache()
+    # Set exactly once from top-level compile
+    is_top_level = False
+    try:
+        if first_ctx and config_cache.saved_config_and_hash is None:
+            is_top_level = True
+            config_cache.saved_config_and_hash = saved_config, saved_config_hash
+            log.debug("Setting top-level compile config hash: %s", saved_config_hash)
+        else:
+            log.debug("Ignoring inner dynamo compile config and hash")  # TODO: remove?
```

Comment:
this should no longer be verbose, so we don't need to remove/put behind verbose flag